job_runner/reporting/event.py

```diff
@@ -87,78 +91,52 @@ def _init_handler(self, msg):
+        }
         if isinstance(msg, Start):
-            self._dump_event(
-                CloudEvent(
-                    {
-                        "type": _FM_JOB_START,
-                        "source": job_path,
-                        "datacontenttype": "application/json",
-                    },
-                    {
-                        "stdout": str(Path(msg.job.std_out).resolve()),
-                        "stderr": str(Path(msg.job.std_err).resolve()),
-                    },
-                )
+            self._dump_cloud_event(
+                attributes={"type": _FM_JOB_START, **job_data},
+                data={
+                    "stdout": str(Path(msg.job.std_out).resolve()),
+                    "stderr": str(Path(msg.job.std_err).resolve()),
+                },
             )
             if not msg.success():
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_FAILURE,
-                            "source": job_path,
-                            "datacontenttype": "application/json",
-                        },
-                        {
-                            "error_msg": msg.error_message,
-                        },
-                    )
+                self._dump_cloud_event(
+                    attributes={"type": _FM_JOB_FAILURE, **job_data},
+                    data={
+                        "error_msg": msg.error_message,
+                    },
                 )
 
         elif isinstance(msg, Exited):
-            if msg.success():
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_SUCCESS,
-                            "source": job_path,
-                        },
-                        None,
-                    )
-                )
-            else:
-                self._dump_event(
-                    CloudEvent(
-                        {
-                            "type": _FM_JOB_FAILURE,
-                            "source": job_path,
-                            "datacontenttype": "application/json",
-                        },
-                        {
-                            "exit_code": msg.exit_code,
-                            "error_msg": msg.error_message,
-                        },
-                    )
-                )
-
-        elif isinstance(msg, Running):
-            self._dump_event(
-                CloudEvent(
+            self._dump_cloud_event(
+                *(
                     {
-                        "type": _FM_JOB_RUNNING,
-                        "source": job_path,
-                        "datacontenttype": "application/json",
-                    },
+                        "type": _FM_JOB_SUCCESS,
+                        "source": job_data["source"],
+                    }
+                    if msg.success()
```

Comment:
I find this a bit harder to read than the previous version, although that is at the cost of some code duplication.

Reply:
I was under the same impression when making the modification. But thought I put it in to see where the disscussion over optimilization vs readability will take me.

Reply:
So what is your thoughts on it now?

Reply:
I think having the ternary operation within the function/method call is a bit too much.
But having multiple calls to the same function/method with similar arguments within a nested block can also be confusing, if not disoriented. So my solution was to reduce nested `if ... else` blocks if possible. With the latest change, I keep the ternary operation, but move it out of the method call and give distict labels that can be understood.